extensions/XeroName/Deltatime.js

```diff
@@ -0,0 +1,223 @@
+// TurboWarp Extension : Deltatime by XeroName
+
+/*
+Referenced articles :
+- https://www.growingwiththeweb.com/2017/12/fast-simple-js-fps-counter.html
+- https://stackoverflow.com/questions/4787431/how-do-i-check-framerate-in-javascript
+- https://developer.mozilla.org/en-US/docs/Web/API/Performance/now
+
+Thanks to the "TheShovel", one of the TurboWarp Extension's developer named "ShovelUtils".
+I learned how to use "Runtime Steps" of Scratch VM through that code.
+*/
+
+(function (Scratch) {
+  'use strict';
+
+  const icon = 'data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjAwIiBoZWlnaHQ9IjYwMCIgdmlld0JveD0iMCAwIDYwMCA2MDAiIGZpbGw9Im5vbmUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CjxjaXJjbGUgY3g9IjMwMCIgY3k9IjMwMCIgcj0iMzAwIiBmaWxsPSIjMjAyMDIwIi8+CjxwYXRoIGQ9Ik04Ny44NjggNTEyLjEzMkM2MC4wMTA0IDQ4NC4yNzQgMzcuOTEyNSA0NTEuMjAzIDIyLjgzNjEgNDE0LjgwNUM3Ljc1OTcyIDM3OC40MDcgLTMuNDQ0MTZlLTA2IDMzOS4zOTcgMCAzMDBDMy40NDQxNmUtMDYgMjYwLjYwMyA3Ljc1OTc0IDIyMS41OTMgMjIuODM2MiAxODUuMTk1QzM3LjkxMjYgMTQ4Ljc5NyA2MC4wMTA0IDExNS43MjYgODcuODY4IDg3Ljg2NzlDMTE1LjcyNiA2MC4wMTA0IDE0OC43OTcgMzcuOTEyNSAxODUuMTk1IDIyLjgzNjFDMjIxLjU5MyA3Ljc1OTcxIDI2MC42MDQgLTkuODYyNjZlLTA2IDMwMCAwQzMzOS4zOTcgOS44NjI2OGUtMDYgMzc4LjQwNyA3Ljc1OTc1IDQxNC44MDUgMjIuODM2MkM0NTEuMjAzIDM3LjkxMjYgNDg0LjI3NSA2MC4wMTA0IDUxMi4xMzIgODcuODY4TDMwMCAzMDBMODcuODY4IDUxMi4xMzJaIiBmaWxsPSIjMzAzMDMwIi8+CjxwYXRoIGQ9Ik0zMzAgNDM1TDIzMCAxODUiIHN0cm9rZT0iIzYxMjM2MSIgc3Ryb2tlLXdpZHRoPSIzMiIgc3Ryb2tlLWxpbmVjYXA9InJvdW5kIi8+CjxwYXRoIGQ9Ik0zMjAgMTg1SDQyME01MjAgMTg1SDQyME00MjAgMTg1VjQzNU0yOTkuNDUxIDQzMy42MjlMMjAwLjkyOCAxODcuMzIxQzIwMC41OTMgMTg2LjQ4MyAxOTkuNDA3IDE4Ni40ODMgMTk5LjA3MiAxODcuMzIxTDEwMC41NDkgNDMzLjYyOUMxMDAuMjg2IDQzNC4yODUgMTAwLjc3IDQzNSAxMDEuNDc3IDQzNUgyOTguNTIzQzI5OS4yMyA0MzUgMjk5LjcxNCA0MzQuMjg1IDI5OS40NTEgNDMzLjYyOVoiIHN0cm9rZT0iIzYwNjA2MCIgc3Ryb2tlLXdpZHRoPSIzMiIgc3Ryb2tlLWxpbmVjYXA9InJvdW5kIi8+CjxwYXRoIGQ9Ik0zMTAgNDE1TDIxMCAxNjUiIHN0cm9rZT0iI0ZGNUNGRiIgc3Ryb2tlLXdpZHRoPSIzMiIgc3Ryb2tlLWxpbmVjYXA9InJvdW5kIi8+CjxwYXRoIGQ9Ik0zMDAgMTY1SDQwME01MDAgMTY1SDQwME00MDAgMTY1VjQxNU0yNzkuNDUxIDQxMy42MjlMMTgwLjkyOCAxNjcuMzIxQzE4MC41OTMgMTY2LjQ4MyAxNzkuNDA3IDE2Ni40ODMgMTc5LjA3MiAxNjcuMzIxTDgwLjU0ODYgNDEzLjYyOUM4MC4yODU4IDQxNC4yODUgODAuNzY5NiA0MTUgODEuNDc3IDQxNUgyNzguNTIzQzI3OS4yMyA0MTUgMjc5LjcxNCA0MTQuMjg1IDI3OS40NTEgNDEzLjYyOVoiIHN0cm9rZT0id2hpdGUiIHN0cm9rZS13aWR0aD0iMzIiIHN0cm9rZS1saW5lY2FwPSJyb3VuZCIvPgo8L3N2Zz4K';
+
+  if (!Scratch.extensions.unsandboxed) {
+    throw new Error('Deltatime by XeroName : This extension must run unsandboxed');
+  }
+
+  const vm = Scratch.vm;
+  const cast = Scratch.Cast;
+
+  let maxFiltStren = 907; // Max value of Filter Strength
+  let filtStren = 8; // Default value of Filter Strength = 8
+  let calcStandFPS = 30; // Standard value of Calculator
+
+  let fT = 0, last = performance.now(), rawFPS, vmFPS, vmDt;
+  const oldStep = vm.runtime._step;
+
+  vm.runtime._step = function () {
+    oldStep.call(this);
```

Comment:
would it make sense to calculate delta time *before* the frame, not after?

Reply:
May I have a time? I'll find a way to fix it.

Reply:
```javascript
let fT = 0, last = performance.now(), rawFPS, vmFPS, vmDt;
  const oldStep = vm.runtime._step;

  vm.runtime._step = function () {
    const now = performance.now();
    let thisFT = now - last;
    fT += (thisFT - fT) / filtStren;

    rawFPS = 1000 / fT;

    if (rawFPS < 0.5) {
      vmFPS = rawFPS.toFixed(2);
    } else {
      vmFPS = Math.round(rawFPS);
    }

    if (vmFPS > 0) {
      vmDt = 1 / vmFPS; // At normal situation
    } else {
      vmDt = 0; // Prevent situation of "DIV/0"
    }
    last = now;

    oldStep.call(this);
  };
```
now would it be make sense? or am I still misleading your point?
always thanks for your kindness.